gui/default/syncthing/core/syncthingController.js

```diff
@@ -912,6 +923,16 @@ angular.module('syncthing.core')
             return '';
         };
 
+        $scope.friendlyNameFromShort = function (shortID) {
+            var matches = $scope.devices.filter(function (n) {
+                return n.deviceID.substr(0, 7) === shortID;
+            });
+            if (matches.length !== 1) {
+                return undefined;
```

Comment:
`return shortID` here.

So we return name if we have the device, short id if we don't, and unknown inf modified by is unset.